### PR TITLE
Fixes #613 Dockerize loklak_search

### LINKED ISSUE #613: Dockerize loklak_search
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**

**Expected behavior:**

Containerize the `loklak_search` with docker
**Screenshots of the issue:**

### PR DESCRIPTION
**Pull request by bhaveshAn**
**Changes proposed in this pull request**
- Dockerize loklak_search
-
-

**Screenshots (if appropriate)** 

**Link to live demo (if appropriate):** 

**Closes #613 **

@jigyasa-grover @mariobehling Please review. Thanks !!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM node:boron
```
Comment (AnshulMalik) [inline Dockerfile]: We can use some light image here https://github.com/sickp/docker-alpine-node

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+MAINTAINER Mario Behling <mb@mariobehling.de>
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+RUN apt-get update && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# install deps
+RUN apt-get install -y --no-install-recommends nodejs=6.9.1 && apt-get clean -y
```
Comment (AnshulMalik) [inline Dockerfile]: If we are already using node image, so is this really required?

**Comment by bhaveshAn** [comment]
@mariobehling Please review

**Comment by harshit98** [comment]
@hemantjadon Please review this pull request. :)

**Comment by bhaveshAn** [comment]
@hemantjadon Your review please on this PR.

**Comment by harshit98** [comment]
@bhaveshAn Share the screenshot of successful deployment of loklak search on Docker.

File: Dockerfile (hunk 1 above)
Comment (singhpratyush) [reply Dockerfile]: I think @AnshulMalik is correct. `node:boron` is 266 MB while `node:boron-alpine` is just 19MB. Using an `alpine` based image would be beneficial.

Comment (bhaveshAn) [reply Dockerfile]: Done @singhpratyush @AnshulMalik . Please review

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM node:boron-alpine
+MAINTAINER Mario Behling <mb@mariobehling.de>
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+RUN apt-get update && apt-get clean && rm -rf /var/lib/apt/lists/*
```
Comment (singhpratyush) [inline Dockerfile]: This won't be needed for an `alpine` based image.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+# copy requirements
+COPY package.json /usr/src/app/
+COPY app.json /usr/src/app/
```
Comment (singhpratyush) [inline Dockerfile]: There is no `app.json`. Are you sure you wanted this here? Please recheck,

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+COPY tslint.json /usr/src/app/
```
Comment (singhpratyush) [inline Dockerfile]: Same here. No `tslint.json` file is present.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+COPY angular-cli.json /usr/src/app/
```
Comment (singhpratyush) [inline Dockerfile]: It should be `.angular-cli.json`.

File: Dockerfile (hunk 5 above)
Comment (bhaveshAn) [reply Dockerfile]: I think it is https://github.com/fossasia/loklak_search/blob/master/tslint.json

**Comment by singhpratyush** [comment]
I am unable to run access the app. Getting following error - 
```js
** NG Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
Date: 2018-01-11T17:33:33.936Z
Hash: 47ea9abd7b3f9d4b39e9
Time: 6170ms
chunk {inline} inline.bundle.js (inline) 5.79 kB [entry] [rendered]
chunk {main} main.bundle.js (main) 3.35 kB [initial] [rendered]
chunk {polyfills} polyfills.bundle.js (polyfills) 981 bytes [initial] [rendered]
chunk {scripts} scripts.bundle.js (scripts) 9.38 kB [initial] [rendered]
chunk {styles} styles.bundle.js (styles) 162 kB [initial] [rendered]
chunk {vendor} vendor.bundle.js (vendor) 853 kB [initial] [rendered]

ERROR in ./src/main.ts
Module build failed: Error: Angular Compiler was detected but it was an instance of the wrong class.
This likely means you have several @ngtools/webpack packages installed. You can check this with `npm ls @ngtools/webpack`, and then remove the extra copies.
    at Object.ngcLoader (/usr/src/app/node_modules/@ngtools/webpack/src/loader.js:458:19)
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
ERROR in ./src/polyfills.ts
Module build failed: Error: Angular Compiler was detected but it was an instance of the wrong class.
This likely means you have several @ngtools/webpack packages installed. You can check this with `npm ls @ngtools/webpack`, and then remove the extra copies.
    at Object.ngcLoader (/usr/src/app/node_modules/@ngtools/webpack/src/loader.js:458:19)
 @ multi ./src/polyfills.ts

webpack: Failed to compile.
```

**Comment by harshit98** [comment]
Similar to https://github.com/fossasia/loklak_search/pull/657. This PR can be closed.

**Pull request closed without merging**